src/main/java/com/wwi21sebgroup5/cinema/services/TmdbService.java

```diff
@@ -0,0 +1,226 @@
+import info.movito.themoviedbapi.model.core.MovieResultsPage;
+import info.movito.themoviedbapi.model.people.PersonCast;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static com.wwi21sebgroup5.cinema.helper.DateFormatter.TMDB_DATE_FORMATTER;
+
+@Service
+public class TmdbService {
+
+    private static final String LANGUAGE = "de-DE";
+    private static final String GERMANY_COUNTRY_CODE = "DE";
+    private static final String DIRECTOR = "Director";
+    private static final String TRAILER = "Trailer";
+    private static final String YOUTUBE = "YouTube";
+    private static final boolean INCLUDE_ADULT = false;
+    private static final int PAGE = 1;
+    private static final int SEARCH_YEAR = -1;
+    private static final String NAME_SPLIT_PATTERN = "\\s+";
+    private static final String BACKDROP_SIZE = "original";
+    private static final String YOUTUBE_BASE_URL = "https://www.youtube.com/watch?v=";
+
+    @Autowired
+    private TmdbApi tmdbApi;
+
+    @Autowired
+    private TmdbConfiguration tmdbConfig;
+
+    @Autowired
+    private ActsInService actsInService;
+
+    @Autowired
+    private DirectorService directorService;
+
+    @Autowired
+    private GenreService genreService;
+
+    @Autowired
+    private ProducerService producerService;
+
+    @Autowired
+    private MovieRepository movieRepository;
+
+    public Movie addMovie(TmdbMovieRequestObject requestObject) throws TmdbMovieNotFoundException,
+            ProducerAlreadyExistsException, DirectorAlreadyExistsException, FSKNotFoundException,
+            TmdbInformationException {
+        Optional<MovieDb> foundMovieDb = this.getMovieById(requestObject.getTmdbMovieId());
+        MovieDb movieDb;
+
+        if (foundMovieDb.isEmpty()) {
+            throw new TmdbMovieNotFoundException(requestObject.getTmdbMovieId());
+        } else {
+            movieDb = foundMovieDb.get();
+        }
+
+        Producer producer = getProducerFromMovieDb(movieDb);
+        Director director = getDirectorFromMovieDB(movieDb);
+        FSK fsk = getFSKfromMovieDb(movieDb);
+        Genre genre = getGenreFromMovieDb(movieDb);
+        String imageUrl = getImageUrlFromMovieDb(movieDb);
+        String trailerUrl = getTrailerUrlFromMovieDb(movieDb);
+        float rating = movieDb.getVoteAverage();
+        String title = movieDb.getTitle();
+        String description = movieDb.getOverview();
+        int length = movieDb.getRuntime();
+        LocalDate startDate = LocalDate.parse(movieDb.getReleaseDate(), TMDB_DATE_FORMATTER);
+
+        Movie newMovie = new Movie(
+                producer, director, fsk, genre, imageUrl, trailerUrl, rating, title, description, length, startDate,
+                null // end date can be set later on, since it depends on the cinema
```

Comment:
Hier eventuell über den Movieservice einen neuen Movie erstellen? Den kann man dann erweitern... Dann ist das Verhalten für alle gleich...
dann werden auch Producer und so weiter einheitlich behandelt

Reply:
Jap, kann man machen. Ich hatte nur in dem PR noch keine Lust den anderen Movie-Service zu refactoren, deswegen gibt es gerade zwei Konstruktoren. Aber kann ich noch umändern